### PR TITLE
Disambiguate indexBy in Query::column()

### PR DESCRIPTION
**Pull request by cebe**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13859

### CODE REVIEW

**Comment by samdark** [comment]
How about https://github.com/yiisoft/yii2/pull/13891?

File: tests/framework/db/ActiveRecordTest.php [hunk 1]
```diff
@@ -1510,6 +1510,7 @@ public function testNoTablenameReplacement()
 
     /**
      * Ensure no ambiguous colum error occurs if ActiveQuery adds a JOIN
```
Comment (samdark) [inline tests/framework/db/ActiveRecordTest.php]: colum → column

File: tests/framework/db/ActiveRecordTest.php [hunk 2]
```diff
@@ -1518,4 +1519,33 @@ public function testAmbiguousColumnFindOne()
         $this->assertTrue($model->refresh());
         CustomerQuery::$joinWithProfile = false;
     }
+
+    /**
+     * Ensure no ambiguous colum error occurs on indexBy with JOIN
```
Comment (samdark) [inline tests/framework/db/ActiveRecordTest.php]: colum → column

**Review by samdark — APPROVED** [review]

**Comment by cebe** [comment]
> How about #13891?

thats unrelated to this bug.

**Pull request closed without merging**